pkg/buildpipeline/pipeline.go

```diff
@@ -69,10 +70,26 @@ type Timeout struct {
 	Unit TimeoutUnit `yaml:"unit,omitempty"`
 }
 
-// TODO: Not yet implemented in build-pipeline
+func (t Timeout) ToDuration() (*metav1.Duration, error) {
+	durationStr := ""
+	// TODO: Populate a default timeout unit, most likely seconds.
```

Comment:
I'd like default values for `workspace` as well, [and apparently its possible by implementing an interface that the marshaller uses](https://github.com/go-yaml/yaml/issues/165), but I couldn't get it working when I tried to do it for `StageOption` specifically (did not spend much time on it though).

Reply:
Yeah, I need to play around with defaulting more.